### DISCUSSION TITLE
citekey formula not working?

### DISCUSSION THREAD

**Discussion by BacZemin** [retorquere/zotero-better-bibtex#2615] [Q&A] [answered]
Hello! I'm a newcomer to this better bibtex. Ultimately, I'm trying to use it as a citewhileyouwrite on obsidian. I was able to install the add-on zotero just fine, but when I try to export (better bibtex), the citekeys are not changing. 
<img width="248" alt="image" src="https://github.com/retorquere/zotero-better-bibtex/assets/123028443/b3b1be35-51f7-4cce-b0d1-94d56f333ec0">

While this is my cite key formula, when I export any format (such as better biblatex) 

all of my citekeys stay default with the unnecessary titles 3,3s such as... @article{alberghinaRedoxControlGlutamine2014}

How do I fix this?

**Comment by retorquere** [comment, marked as answer]
The citekeys are not expected to change on export. If you change the citekey formula and you want it applied to existing items, you must select those items, right click, and choose Better BibTeX -> Refresh from the popup menu.

**Reply by BacZemin** [reply]
Thank you so much!!! it works.

**Reply by Sciroccogti** [reply]
Thanks a lot ! I've been stuck on this for hours, maybe I have missed some notation somewhere in the document...

**Reply by retorquere** [reply]
@Sciroccogti I don't know what you are asking.

**Reply by Sciroccogti** [reply]
There's no question, I just don't know how to make the formula to work until I see this discussion.

**Comment by klasje1** [comment]
I am having this same problem with Zotero integration in Obsidian. I change the citekey formula in Zotero and then export to Obsidian, but it doesn't change from the default

**Reply by BacZemin** [reply]
this problem was resolved. If you refresh it, it works.

**Reply by klasje1** [reply]
Yeah, it works that way. Do you know how to update the "Active citation key formula" below the Citation key formula window? 
I have changed the citation key formula, but now the active citation won't change by itself.

**Reply by retorquere** [reply]
You're not meant to update that, it is for display only. While you're editing the formula you are inevitably going to have times when what you have typed so far is not a valid formula; as long as what you've typed so far isn't a valid formula, BBT shows there what the formula is that BBT actively uses, which is the last valid formula it knows about. As soon as what you type is an valid formula, it becomes the active formula, and then that 2nd display disappears, since there's no discrepancy between them.

I'm not sure what you mean by "active citation won't change by itself", can you elaborate?

**Reply by klasje1** [reply]
I fixed it with what you said about the formula being not valid. I had a typo in the formula and now is working and exporting with the format that I wanted.

I was just confused about it. Thank you!!
